### PR TITLE
feat: strict hooks values

### PR DESCRIPTION
**Pull request by alexnm**
Replaces `useSubscription` with two separate tooks:
* `useWorkspaceSubscription` holds the values for free/pro flags, provider + subscription object
* `useWorkspaceLimits` holds all the flags and numbers related to the limitations

I went ahead and replaced all `!<something>` conditions with the explicit true check for `isPro`/`isFree`.

The hooks now follow a similar approach in v1/v2 where we treat the initial value as `undefined` so there's no risk of flickering elements on the screen as long as we use the true values to conditionally render.

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/7180/commits/ee5b4fdf7a97d725ab3c4f20b50ea39cf3ee7535 is at https://pr7180.build.csb.dev/s/new.

File: .codesandbox/tasks.json [hunk 1]
```diff
@@ -14,9 +27,6 @@
       "runAtStart": true,
       "preview": {
         "port": 3000
-      },
-      "restartOn": {
-        "files": ["yarn.lock", "packages/app/scripts/*"]
```
Comment (alexnm) [inline .codesandbox/tasks.json:19]: wow, need to fix this, we have a bug with generating the task config, probably it omits the restartOn

File: packages/app/src/app/constants.ts [hunk 2]
```diff
@@ -8,3 +8,9 @@ export const SUBSCRIPTION_DOCS_URLS = {
       '/docs/learn/introduction/workspace#managing-teams-and-subscriptions',
   },
 };
+
+// Soft limit of maximum amount of pro
+// editor a team can have. Above this,
+// we should prompt CTAs to enable
+// custom pricing.
+export const MAX_PRO_EDITORS = 20;
```
Comment (alexnm) [inline packages/app/src/app/constants.ts:16]: moved from useSubscription.ts

File: packages/app/src/app/hooks/useWorkspaceAuthorization.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export const useWorkspaceAuthorization = () => {
    * Team states
    */
 
-  const isTeamSpace = !isPersonalSpace;
+  const isTeamSpace = activeTeam !== null && !isPersonalSpace;
```
Comment (alexnm) [inline packages/app/src/app/hooks/useWorkspaceAuthorization.ts:21]: isTeamSpace should be true only after the team is loaded (unfortunately null means the team is not yet there in this case)

File: packages/app/src/app/pages/Dashboard/Components/SyncedSandbox/index.tsx [hunk 4]
```diff
@@ -59,6 +59,7 @@ export const SyncedSandbox = ({
     onDoubleClick,
     onContextMenu,
     onBlur,
+    isViewOnly: false,
```
Comment (alexnm) [inline packages/app/src/app/pages/Dashboard/Components/SyncedSandbox/index.tsx:62]: random typecheck error

File: .codesandbox/tasks.json [hunk 5]
```diff
@@ -31,16 +44,10 @@
     "storybook": {
       "name": "Run storybook",
       "command": "cd packages/components && yarn start:storybook"
-    }
-  },
-  "setupTasks": [
```
Comment (alexnm) [inline .codesandbox/tasks.json:36]: setupTasks are moved at the beginning when the config is generated

File: .codesandbox/tasks.json [hunk 6]
```diff
@@ -1,4 +1,17 @@
 {
+  // These tasks will run in order when initializing your CodeSandbox project.
```
Comment (tristandubbeld) [inline .codesandbox/tasks.json:2]: Comments in `json`, does this not throw any errors? 👀

File: .codesandbox/tasks.json (hunk 1 above)
Comment (tristandubbeld) [reply .codesandbox/tasks.json]: Is this something that we need to address somewhere or is it limited to this repository?

**Review by tristandubbeld — APPROVED** [review]
Great work Alex, readable and explicit! Walked through the application a bit and didn't notice any weird popping banners or stripes.

File: .codesandbox/tasks.json (hunk 1 above)
Comment (alexnm) [reply .codesandbox/tasks.json]: created a ticket for the editor team, it's a pitcher problem

File: .codesandbox/tasks.json (hunk 6 above)
Comment (alexnm) [reply .codesandbox/tasks.json:2]: pitcher removes comments before deserializing the config file. these have been added a while ago to help users navigate the json file better

Comment (tristandubbeld) [reply .codesandbox/tasks.json:2]: ~Shouldn't it be a `.jsonc` file then?~ [Doesn't look like it needs to be an explicit `.jsonc` file](https://code.visualstudio.com/docs/languages/json#_json-with-comments). So I guess we're good.

File: packages/app/src/app/pages/Dashboard/Components/Selection/ContextMenus/SandboxMenu.tsx [hunk 7]
```diff
@@ -179,7 +181,7 @@ export const SandboxMenu: React.FC<SandboxMenuProps> = ({
               sandboxIds: [item.sandbox.id],
               preventSandboxLeaving:
                 item.sandbox.permissions.preventSandboxLeaving ||
-                (!hasActiveSubscription && hasMaxPublicSandboxes),
+                (isFree && hasMaxPublicSandboxes),
```
Comment (olarclara) [inline packages/app/src/app/pages/Dashboard/Components/Selection/ContextMenus/SandboxMenu.tsx:184]: Is `isFree` redundant here?

**Review by olarclara — APPROVED** [review]
Looks good!

But I think we can remove all the `isFree && hasSomeLimit` because these values get computed already taking `isFree` into account.

**Pull request merged**